prometheus/common/alert-rules.d/crunchy-alert-rules-pg.yml.example

```diff
@@ -56,21 +56,16 @@ groups:
     annotations:
       summary: '{{ $labels.job }} has had a PostgreSQL failover event. Please check systems involved in this cluster for more details'
 
-## Whether a system switches from primary to replica or vice versa must be configured per named job.
-## No way to tell what value a system is supposed to be without a rule expression for that specific system
-## 2 to 1 means it changed from primary to replica. 1 to 2 means it changed from replica to primary
-## Set this alert for each system that you want to monitor a recovery status change
-## Below is an example for a target job called "Replica" and watches for the value to change above 1 which means it's no longer a replica
-#
-#  - alert: PGRecoveryStatusSwitch_Replica
-#    expr: ccp_is_in_recovery_status{job="Replica"} > 1
-#    for: 60s
-#    labels:
-#      service: postgresql
-#      severity: critical
-#      severity_num: 300
-#    annotations:
-#      summary: '{{ $labels.job }} has changed from replica to primary'
+## Role switch from Replica to Primary
+
+  - alert: PGRecoveryStatusSwitch
+    expr: increase(ccp_is_in_recovery_status[60s]) > 1    
```

Comment:
Is this doing anything different than the rule previous to this one (PGRecoveryStatusSwitch) that detects any role switch (primary -> replica or replica -> primary) other than only detecting a replica coming out of recovery?

Reply:
The rule which was replaced was hardcoded to the specific target (job).  This means you needed to designate certain targets (jobs) as being replicas.  Since there is no preference on which target is replica or primary, it made it difficult to use the existing rule to alert when a promotion occurred.  This new rule will alert whenever a replica is promoted to primary indicating that a failover has occurred.

Reply:
Right. That's what the other rule, that is enabled by default, is doing. It just handles things in both directions.

Reply:
Yes, the other rule does alert in both directions, but only concerned about 1 promotion.

Reply:
So I tested this out and with a `60s` window it fires and resolves very quickly (less than a minute). I changed it to a `5m` window like the existing rule does and it sticks around a bit longer to at least give someone some time to check on the alert to see what it's doing.

Since this one also auto-resolves like the other alert, I'm not sure I see a need for a different one. The purpose of the rule you replaced is for creating a rule that does not self-resolve, which may be desirable for some users that don't have auto-failover configured. So I do not want to remove that example either.

If we changed to having the rule only alert on replica -> primary, we'd potentially miss situations where patroni can put an entire cluster into read-only mode when it cannot perform a leader election. So we'd then also want to have another  rule watching for primary -> replica. 

Since the existing rule covers both of these scenarios, I'd be inclined to just use that one as it is now for the default rules we provide.

Reply:
I could put an additional comment on the manual rule that it is most commonly not necessary if the users have the other auto-resolving alert configured

Reply:
True, but that is why we added the PGNoPrimary rule.

Reply:
Auto resolve is ok as we only want the alert to trigger a notification and capture the event in history.